tests/workgroup/run_kernel.c

```diff
@@ -66,6 +67,10 @@ main (int argc, char **argv)
   size_t local_work_size[3];
   char kernel_path[2048];
 
+  /* Use the single thread cpu-minimal by default to get guaranteed work-group
+     execution order for stable global id printouts, unless overriden from
+     the environment. */
+  setenv ("POCL_DEVICES", "cpu-minimal", 0);
```

Comment:
I don't think putting environment configuration into test code is a good practice. I propose a different solution: if the tests needs stable execution order, then instead of messing with POCL_DEVICES it should check for number of compute-units and return a skip code if it's larger than one. Then it can work with any device that has 1 CU.

Reply:
The problem is how to enforce such a device without forcing it always. Also, even with 1CU the WG execution order is not guaranteed. Not sure what's best here.